NuRadioReco/modules/channelReadoutWindowCutter.py

```diff
@@ -158,21 +158,27 @@ def _get_number_of_samples(sampling_rate, detector_sampling_rate, detector_n_sam
     # the number of samples after resampling will be correct
     valid_sampling_rate = sampling_rate % detector_sampling_rate < 1e-8
 
-    if issue_warning and not valid_sampling_rate:
-            logger.warning(
-                'The current sampling rate '
-                f'({sampling_rate/units.GHz:.3f} GHz) is not a multiple of '
-                f'the target detector sampling rate ({detector_sampling_rate/units.GHz:.3f} GHz). '
-                'Traces may not have the correct trace length after resampling.'
-            )
-
     # this should ensure that 1) the number of samples is even and
     # 2) resampling to the detector sampling rate results in the correct number of samples
     # (note that 2) can only be guaranteed if the detector sampling rate is lower than the
     # current sampling rate)
     number_of_samples = int(
         2 * np.ceil(detector_n_samples / 2 * sampling_rate / detector_sampling_rate))
 
+    if not valid_sampling_rate:
+        # actually check if the number of samples is correct after resampling
+        n_final = len(signal_processing.resample(np.zeros(number_of_samples), detector_sampling_rate / sampling_rate))
+        valid_sampling_rate = n_final == detector_n_samples
+
+        if not valid_sampling_rate and issue_warning:
+            logger.warning(
+                'The current sampling rate '
+                f'({sampling_rate/units.GHz:.3f} GHz) is not a multiple of '
+                f'the target detector sampling rate ({detector_sampling_rate/units.GHz:.3f} GHz). '
+                f'Traces may not have the correct trace length after resampling. Desired number of samples: {detector_n_samples}, '
+                f'expected number of samples after resampling: {n_final}.'
+            )
```

Comment:
The first part of this `if` statement is now redundant. I would also probably change the warning (maybe even bump it up to logging.error?) and make it say that the trace **will not** have the correct number of samples after resampling, because now you actually check this. And yes, the reason that I hadn't implemented this before was because now you do resample twice instead of once, though probably the performance impact is something we can live with (could consider writing a private function that returns `n_final` and decorating it with lru_cache, but this is probably overengineering things.

Reply:
The first part of the if statement is not redundant. After the the first if condition `valid_sampling_rate` is updated. I changed the code to an error and added a function with an lru_cache.